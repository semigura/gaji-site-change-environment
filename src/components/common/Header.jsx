import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'gatsby';
import Icon from './icons/Icon';

import css from './Header.module.scss';

const GLOBAL_NAVIGATION = [
  {
    label: 'お知らせ',
    to: '/news/',
  },
  {
    label: 'サービス案内',
    to: '/services/',
  },
  {
    label: '事例紹介',
    to: '/case-studies/',
  },
  {
    label: '会社案内',
    to: '/about/',
  },
  {
    label: 'ブログ',
    to: 'https://blog.gaji.jp/',
  },
  {
    label: 'お問い合わせ',
    to: '/contact/',
  },
];

class Header extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle() {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  render() {
    const { className } = this.props;
    const { isOpen } = this.state;
    const classes = classNames(css.Header, className, {
      [css.isOpen]: isOpen,
    });
    const menuClasses = classNames(css.Header__Menu, {
      [css.isOpen]: isOpen,
    });
    const isNode =
      typeof process !== 'undefined' && process.title !== 'browser';
    const pathname = isNode ? undefined : window.location.pathname;
    const linkClasses = to => {
      const isActive = pathname
        ? pathname.match(new RegExp(`^${to}`))
        : undefined;
      return classNames(css.Header__Link, { [css.isActive]: isActive });
    };
    const LogoTag = pathname === '/' ? 'h1' : 'span';
    const icon = isOpen
      ? { label: 'Close', name: 'close' }
      : { label: 'Open menu', name: 'menu' };

    return (
      <div className={classes}>
        <header className={css.Header__Inner}>
          <Link className={css.Header__Logo} to="/">
            <LogoTag className={css.Header__LogoInner}>
              <Icon
                className={css.Header__Icon}
                name="logo"
                aria-label="Gaji-Labo（ガジラボ）"
              />
            </LogoTag>
          </Link>
          <nav className={css.Header__Nav}>
            <button
              aria-label={icon.label}
              className={css.Header__Button}
              onClick={this.handleToggle}
              type="button"
            >
              <Icon name={icon.name} />
            </button>
            <ul className={menuClasses}>
              {GLOBAL_NAVIGATION.map(nav => (
                <li className={css.Header__Item} key={nav.to}>
                  {nav.to === 'https://blog.gaji.jp/' ? (
                    <a
                      className={linkClasses(nav.to)}
                      href={nav.to}
                    >
                      {nav.label}
                    </a>
                  ) : (
                    <Link
                      className={linkClasses(nav.to)}
                      activeClassName={css.isActive}
                      to={nav.to}
                    >
                      {nav.label}
                    </Link>
                  )}
                </li>
              ))}
            </ul>
          </nav>
        </header>
      </div>
    );
  }
}

Header.propTypes = {
  className: PropTypes.string,
};

Header.defaultProps = {
  className: undefined,
};

export default Header;
