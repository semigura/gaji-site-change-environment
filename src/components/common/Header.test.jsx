import { shallow } from 'enzyme';
import React from 'react';

import Header from './Header';

describe('<Header />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleToggle', () => {
    const wrapper = shallow(<Header />);
    const button = wrapper.find('.Header__Button');

    it('renders correctly', () => {
      button.simulate('click');
      const menu = wrapper.find('.Header__Menu');
      expect(menu.hasClass('isOpen')).toBeTruthy();
    });
  });
});
