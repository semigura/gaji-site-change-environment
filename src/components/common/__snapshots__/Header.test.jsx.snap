// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<div
  className="Header"
>
  <header
    className="Header__Inner"
  >
    <mockConstructor
      className="Header__Logo"
      to="/"
    >
      <span
        className="Header__LogoInner"
      >
        <Icon
          aria-label="Gaji-Labo（ガジラボ）"
          className="Header__Icon"
          name="logo"
        />
      </span>
    </mockConstructor>
    <nav
      className="Header__Nav"
    >
      <button
        aria-label="Open menu"
        className="Header__Button"
        onClick={[Function]}
        type="button"
      >
        <Icon
          name="menu"
        />
      </button>
      <ul
        className="Header__Menu"
      >
        <li
          className="Header__Item"
          key="/news/"
        >
          <mockConstructor
            activeClassName="isActive"
            className="Header__Link"
            to="/news/"
          >
            お知らせ
          </mockConstructor>
        </li>
        <li
          className="Header__Item"
          key="/services/"
        >
          <mockConstructor
            activeClassName="isActive"
            className="Header__Link"
            to="/services/"
          >
            サービス案内
          </mockConstructor>
        </li>
        <li
          className="Header__Item"
          key="/case-studies/"
        >
          <mockConstructor
            activeClassName="isActive"
            className="Header__Link"
            to="/case-studies/"
          >
            事例紹介
          </mockConstructor>
        </li>
        <li
          className="Header__Item"
          key="/about/"
        >
          <mockConstructor
            activeClassName="isActive"
            className="Header__Link"
            to="/about/"
          >
            会社案内
          </mockConstructor>
        </li>
        <li
          className="Header__Item"
          key="https://blog.gaji.jp/"
        >
          <a
            className="Header__Link"
            href="https://blog.gaji.jp/"
          >
            ブログ
          </a>
        </li>
        <li
          className="Header__Item"
          key="/contact/"
        >
          <mockConstructor
            activeClassName="isActive"
            className="Header__Link"
            to="/contact/"
          >
            お問い合わせ
          </mockConstructor>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
